doc/design/vic-machine/update.md

```diff
@@ -0,0 +1,207 @@
+# VCH Update (P0)
+This document defines the VCH update command line, and the related changes in vic-machine, to make VCH update process more reasonable.
+
+Here is the key points considerred in this Definition
+- Batch Operation
+
+  We'd like to update VCH configurations more than one item at one time to make the command easy to use
+- Reuse all options used in vic-machine create
+
+  In vic-machine create, we already have 57 options to define VCH configurations, though not all of them can be updated, quite amount of them should be able to be changed, and this number will grow up over time, so we definitely do not want to introduce 3X that number of options to update (update means add/modify/remove)
+
+## Options
+First option is to introduce update primitives
+
+### Delta Options
+#### Update Primitives
+
+- add
+- modify
+- remove, rm
+
+In one update command, only one delta primitive is acceptable, and only that kind of operation will be executed.
+
+#### Example Command
+
+- vic-machine update --add --volume-store ... --container-network ...
+
+  add additional volume-store and container-network into VCH
+- vic-machine update --modify --dns-server ... --thumbprint ... --momery ...
+
+  update dns-server, target thumbprint and resource pool memory in the VCH
+- vic-machine update --rm --volume-store ...  --dns-server ...
+
+  Remove one volume-store and remove custom dns-server from VCH
+
+We do not allow command like following
+- vic-machine update --add --volume-store ... --container-network ... --rm --volume-store ...  --dns-server ...
+
+In this way, the update command can only run one kind of change at one time
+
+### Whole Thing Update
+Other than introducing delta primitives, we could also require user to specify all configurations during update. That means the old VCH configuration will be replaced completely by this new configuration.
+
+Anything specified in the new configuration, which does not exist in old configuration will be added, anything missing in new configuration will be removed. And for the configuration can only have one item, will be replaced.
```

Comment:
This is essentially what I described in my comment on the update primitives, however by dropping delete it means you only specify the deltas, not the entire thing.